### PR TITLE
Subcell: Add TciOnFdGrid mutator for NewtonianEuler 

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Add the troubled-cell indicator on the FD grid for NewtonianEuler.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * Computes the primitive variables on the DG and subcell grids, mutating the
+ * subcell/active primitive variables in the DataBox. Then, checks that the
+ * minimum density and pressure on both the DG and subcell mesh are above
+ * \f$10^{-18}\f$, and runs the Persson TCI on the density and pressure on the
+ * DG grid. The reason for applying the Persson TCI to both the density and
+ * pressure is to flag cells at contact discontinuities.
+ *
+ * Please note that the TCI is run after the subcell solution has been
+ * reconstructed to the DG grid, and so `Inactive<Tag>` is updated DG solution.
```
Comment (kidder) [inline src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp]: is the updated

File: src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      tmpl::list<MassDensity, Velocity, SpecificInternalEnergy, Pressure>>>;
+  using argument_tags =
+      tmpl::list<MassDensityCons, MomentumDensity, EnergyDensity,
+                 Inactive<MassDensityCons>, Inactive<MomentumDensity>,
+                 Inactive<EnergyDensity>, hydro::Tags::EquationOfStateBase,
+                 domain::Tags::Mesh<Dim>>;
+
+  template <size_t ThermodynamicDim>
+  static bool apply(
+      gsl::not_null<Variables<
```
Comment (kidder) [inline src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp:81]: forward declare Variables

File: tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TciOnFdGrid.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    CHECK(result);
+  }
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.NewtonianEuler.Subcell.TciOnFdGrid",
+                  "[Unit][Evolution]") {
+  for (const auto test_this :
+       {TestThis::AllGood, TestThis::SmallDensityDg,
+        TestThis::SmallDensitySubcell, TestThis::PerssonDensity}) {
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TciOnFdGrid.cpp]: add tests for pressure triggers?

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

File: src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      dg_momentum_density, dg_energy_density, eos);
+
+  using std::min;
+  return min(min(get(subcell_mass_density)), min(get(dg_mass_density))) <
+             min_density_allowed or
+         min(min(get(get<Pressure>(*subcell_grid_prim_vars))),
+             min(get(get<Pressure>(dg_grid_prim_vars)))) <
+             min_pressure_allowed or
+         evolution::dg::subcell::persson_tci(dg_mass_density, dg_mesh,
+                                             persson_exponent, 1.0e-18) or
```
Comment (fmahebert) [inline src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.cpp]: Can you make a `constexpr double persson_tci_epsilon = 1.0e-18;` as in the other TCI?

File: src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+/*!
+ * \brief Troubled-cell indicator applied to the finite difference subcell
+ * solution to check if the corresponding DG solution is admissible.
+ *
+ * Computes the primitive variables on the DG and subcell grids, mutating the
+ * subcell/active primitive variables in the DataBox. Then, checks that the
+ * minimum density and pressure on both the DG and subcell mesh are above
+ * \f$10^{-18}\f$, and runs the Persson TCI on the density and pressure on the
+ * DG grid. The reason for applying the Persson TCI to both the density and
+ * pressure is to flag cells at contact discontinuities.
```
Comment (fmahebert) [inline src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp]: Update this to match the newer text in the other TCI

Also, can you explain why you check both grids in a function "TciOnFdGrid" ?

**Comment by nilsdeppe** [comment]
I've rebased and pushed a fixup. Thanks for the reviews!

File: src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp [hunk 6]
```diff
@@ -30,14 +32,21 @@ namespace NewtonianEuler::subcell {
...
- * pressure is to flag cells at contact discontinuities.
+ * subcell/active primitive variables in the DataBox. Then,
+ * - if the minimum density or pressure on either the DG or subcell mesh are
+ *   below \f$10^{-18}\f$, marks the element as troubled and returns. We check
+ *   both the FD and DG grids since when a discontinuity is inside the element
+ *   oscillations in the DG solution can result in negative values that aren't
+ *   present in the FD solution.
+ * - runs the Persson TCI on the density and pressure on the DG grid. The reason
+ *   for applying the Persson TCI to both the density and pressure is to flag
+ *   cells at contact discontinuities. The Persson TCI is only works with
```
Comment (kidder) [inline src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp]: is only -> only

**Review by kidder — CHANGES_REQUESTED** [review]
you have my permission to squash (including the one additional requested change)

**Comment by fmahebert** [comment]
LGTM, please squash

**Comment by nilsdeppe** [comment]
I've squashed. Thanks for reviewing :)

**Review by kidder — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Pull request merged**